autoPyTorch/pipeline/image_classification.py

```diff
@@ -134,7 +136,9 @@ def _get_hyperparameter_search_space(self,
         """
         cs = ConfigurationSpace()
 
-        if dataset_properties is None or not isinstance(dataset_properties, dict):
+        if not isinstance(dataset_properties, dict):
+            warnings.warn('The given dataset_properties argument contains an illegal value.'
```

Comment:
`illegal value` is a common usage in python?
I have never seen `illegal value`, but see often `invalid value`.

Plus, `The given dataset_properties argument contains ..` sounds we have weird values inside the dict rather than `dataset_properties` is not a valid type.
So I think `dataset_properties must be dict` is easier to understand the warning.